libr/core/cmd_debug.c

```diff
@@ -2714,6 +2714,81 @@ static void asciiart_backtrace(RCore *core, RList *frames) {
 	}
 }
 
+
+static void get_backtrace_info(RCore* core, RDebugFrame* frame, ut64 addr, char * flagdesc, char * flagdesc2, char *pcstr, char* spstr, bool hex_format) {
+	int sizeof_flags = 1024;
+	int sizeof_str = 32;
+	RFlagItem *f = r_flag_get_at (core->flags, frame->addr, true);
+	flagdesc[0] = flagdesc2[0] = 0;
+	if (f) {
+		if (f->offset != addr) {
+			int delta = (int)(frame->addr - f->offset);
+			if (delta > 0) {
+				snprintf (flagdesc, sizeof_flags,
+						"%s+%d", f->name, delta);
+			} else if (delta < 0) {
+				snprintf (flagdesc, sizeof_flags,
+						"%s%d", f->name, delta);
+			} else {
+				snprintf (flagdesc, sizeof_flags,
+						"%s", f->name);
+			}
+		} else {
+			snprintf (flagdesc, sizeof_flags,
+					"%s", f->name);
+		}
+	}
+	f = r_flag_get_at (core->flags, frame->addr, true);
+	if (f && !strchr (f->name, '.')) {
+		f = r_flag_get_at (core->flags, frame->addr - 1, true);
+	}
+	if (f) {
+		if (f->offset != addr) {
+			int delta = (int)(frame->addr - 1 - f->offset);
+			if (delta > 0) {
+				snprintf (flagdesc2, sizeof_flags,
+						"%s+%d", f->name, delta + 1);
+			} else if (delta<0) {
+				snprintf (flagdesc2, sizeof_flags,
+						"%s%d", f->name, delta + 1);
+			} else {
+				snprintf (flagdesc2, sizeof_flags,
+						"%s+1", f->name);
+			}
+		} else {
+			snprintf (flagdesc2, sizeof_flags,
+					"%s", f->name);
+		}
+	}
+	if (!strcmp (flagdesc, flagdesc2)) {
+		flagdesc2[0] = 0;
+	}
+
+	if (hex_format) {
+		if (core->dbg->bits & R_SYS_BITS_64) {
+			snprintf (pcstr, sizeof_str, "0x%-16" PFMT64x, frame->addr);
+			snprintf (spstr, sizeof_str, "0x%-16" PFMT64x, frame->sp);
+		} else if (core->dbg->bits & R_SYS_BITS_32) {
+			snprintf (pcstr, sizeof_str, "0x%-8" PFMT64x, frame->addr);
```

Comment:
any reason for `%-` ?

Reply:
I copied it from the code, but I guess its to align the addresses in the print